airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/paginators/strategies/offset_increment.py

```diff
@@ -16,14 +18,17 @@ class OffsetIncrement(PaginationStrategy, JsonSchemaMixin):
     Pagination strategy that returns the number of records reads so far and returns it as the next page token
 
     Attributes:
-        page_size (int): the number of records to request
+        page_size (InterpolatedString): the number of records to request
     """
 
-    page_size: int
+    config: Config
+    page_size: Union[InterpolatedString, str, int]
     options: InitVar[Mapping[str, Any]]
 
     def __post_init__(self, options: Mapping[str, Any]):
         self._offset = 0
+        self.page_size.string = str(self.page_size.string)
```

Comment:
this won't work if the page_size isn't already an InterpolatedString. It's a little annoying to do, but you'll need to check the type of the field and update it accordingly

Reply:
I have tested it with three value types and it works without errors. As I see we always have `self.page_size` as interpolated string type. The difference is in `.string` attribute which has int/str/ type depending on value in .yaml config. This was test value in my config:
int:
```
pagination_strategy:
      type: OffsetIncrement
      page_size: 2
```
str
```
pagination_strategy:
      type: OffsetIncrement
      page_size: "2"
```
Interpolated string
```
pagination_strategy:
      type: OffsetIncrement
      page_size: "{{ options['items_per_page'] }}"
```